### PR TITLE
Return plain format fx rate

### PR DESCRIPTION
**Pull request by yunshine**
## Context

Addresses [this issue in #14823](https://github.com/degica/hats/pull/14823#discussion_r955555300)

Some currencies, like USD and EUR, have subunits (ie. they use cents). So, when doing foreign exchange conversion, $1 should be written out as 100 USD, not 1 USD.

In order to provide an accurate foreign exchange rate for currencies with subunits, this gem moves the decimal point of an exchange rate depending on the base currency's subunit. However, there are situations when the user may not want an exchange rate with the decimal point of the exchange rate moved. For example, on show pages where 100 USD is displayed as $1.00, you'd want display an exchange rate that doesn't have the decimal place moved since the currency is being displayed as a decimal.

This PR allows this gem to return an exchange rate that doesn't have its decimal place moved.

## How to test

1. In your terminal, run `bin/console`
2. This repo is public, so I won't be providing any secret credentials in these instructions... sorry! You'll have to go to 1Password, look for Openexchangerates, visit this [link](https://openexchangerates.org/account/app-ids), sign in, and get the `app_id` which will be used in step 3.
3. Configure the gem by running the following. Don't forget to add the Open Exchange Rates `app_id` that you just got in step 2!
`OmniExchange.configure do |config|
  config.provider_config = {
    open_exchange_rates: {
      app_id: "THE APP ID GOES HERE"
    }
  }
end`
4. Use the gem to convert 100 USD ($1) to JPY - `usd_to_jpy = OmniExchange.get_fx_data(amount: 100, base_currency: "USD", target_currency: "JPY", providers: [:open_exchange_rates])`
5. Take note of the converted amount - `test1 = usd_to_jpy[:converted_amount].to_f`
6. Take note of the converted amount when converting $1 using 100 and the old version of the exchange rate - `test2 = 100 * (usd_to_jpy[:exchange_rate].to_f)`
7. Take note of the converted amount when converting $1 using 1.00 and the new version of the exchange rate - `test3 = 1.00 * (usd_to_jpy[:non_subunit_fx_rate].to_f)`
8.  Please check to make sure that all three converted amounts are the same.
image

### CODE REVIEW

**Comment by degicat** [comment]
@Agroang can you help us review this PR, please?

**Comment by degicat** [comment]
@barom1991 can you help us review this PR, please?

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,6 @@ omni_exchange-1.1.0.gem
 omni_exchange-1.1.1.gem
 omni_exchange-1.2.0.gem
 omni_exchange-1.3.0.gem
+omni_exchange-1.4.0.gem
+omni_exchange-1.5.0.gem
+omni_exchange-1.6.0.gem
```
Comment (Metallion) [inline .gitignore]: If gitignore supports wild cards this might be a good place to use them :D

Comment (yunshine) [reply .gitignore]: @Metallion I didn't even know that was a thing! I'll look into it... thank you for the suggestion!

Comment (barom1991) [reply .gitignore:25]: I think `*.gem` is fine.
By the way, should we ignore this file? It is not generated in my environment.

Comment (barom1991) [reply .gitignore]: I think `*.gem` is fine.
By the way, should we ignore this file? It is not generated in my environment.

**Review by barom1991 — COMMENTED** [review]
Had a small comment. it seems to work fine 🙆‍♂️

**Comment by yunshine** [comment]
@barom1991 Thanks for the review and the `*.gem`! 🙇🏻 And I think you're right... there really isn't a reason to ignore the .gem files. I've unignored them in [9970c8a](https://github.com/degica/omni_exchange/pull/4/commits/9970c8ab6984e944e3ae5763c41f0c5720794875).

**Comment by barom1991** [comment]
@yunshine 
I think we can delete the .gem file.
The .gem file is not generated in my environment, but if it is generated again, I think it is OK to add it to `.gitignore`.

**Comment by yunshine** [comment]
> @yunshine I think we can delete the .gem file. The .gem file is not generated in my environment, but if it is generated again, I think it is OK to add it to `.gitignore`.

@barom1991 Thanks for the clarification. Done!

**Review by barom1991 — APPROVED** [review]
LGTM 👍

**Pull request merged**